### PR TITLE
More ETL removals, including Table

### PR DESCRIPTION
**Pull request by calina-c**
Contains #1097, but more radical. This PR also removes Table, which has become redundant. Without a table registry, we have no use with Table objects.

Table = NamedTable that contains type + PPSS configuration

PPSS configuration comes always from the data factory, so we can inject it more freely without creating connective objects such as Table.

### CODE REVIEW

File: pdr_backend/analytics/test/test_get_predictions_info.py [hunk 1]
```diff
@@ -31,8 +31,8 @@ def test_get_predictions_info_main_mainnet(
         fin_timestr=fin_timestr,
     )
     predictions_df = _gql_datafactory_first_predictions_df
-    predictions_table = Table(Prediction, ppss)
-    predictions_table.append_to_storage(predictions_df)
+    predictions_table = NamedTable.from_dataclass(Prediction)
```
Comment (idiom-bytes) [inline pdr_backend/analytics/test/test_get_predictions_info.py:34]: yeah, this is already so much better...

File: pdr_backend/lake/test/conftest.py [hunk 2]
```diff
@@ -601,16 +601,16 @@ def setup_data(
...
     }
 
-    gql_tables["pdr_predictions"].append_to_storage(preds)
-    gql_tables["pdr_truevals"].append_to_storage(truevals)
-    gql_tables["pdr_payouts"].append_to_storage(payouts)
-    gql_tables["pdr_slots"].append_to_storage(slots)
+    gql_tables["pdr_predictions"].append_to_storage(preds, ppss)
+    gql_tables["pdr_truevals"].append_to_storage(truevals, ppss)
+    gql_tables["pdr_payouts"].append_to_storage(payouts, ppss)
+    gql_tables["pdr_slots"].append_to_storage(slots, ppss)
```
Comment (idiom-bytes) [inline pdr_backend/lake/test/conftest.py:613]: I'm seeing ppss being passed around so much.
It's growing too.
And it's a configuration file.

Isn't there a better way?

File: pdr_backend/lake/csv_data_store.py [hunk 3]
```diff
@@ -123,8 +123,8 @@ def __init__(self, base_path: str, table_name: str):
         self.table_name = table_name
 
     @staticmethod
-    def from_table(table):
-        return CSVDataStore(table.base_path, table.table_name)
+    def from_table(table, ppss):
+        return CSVDataStore(ppss.lake_ss.lake_dir, table.table_name)
```
Comment (idiom-bytes) [inline pdr_backend/lake/csv_data_store.py:127]: I'm seeing ppss being passed around so much.
It's growing too.
And it's a configuration file.

Isn't there a better way?

File: pdr_backend/lake/gql_data_factory.py [hunk 4]
```diff
@@ -113,11 +77,14 @@ def _prepare_temp_table(self, dataclass: Type[LakeMapper], st_ut, fin_ut):
             # 3. in preparation to append, check missing data to move FROM CSV -> TO TEMP TABLES
             # 4. resume appending to CSVs + Temp tables until complete
         """
-        table = Table(dataclass, self.ppss)
+        table = NamedTable.from_dataclass(dataclass)
+        temp_table = TempTable.from_dataclass(dataclass)
```
Comment (idiom-bytes) [inline pdr_backend/lake/gql_data_factory.py:81]: :+1:

File: pdr_backend/lake/gql_data_factory.py [hunk 5]
```diff
@@ -256,7 +225,7 @@ def _do_subgraph_fetch(
             pagination_offset += pagination_limit
 
         if len(buffer_df) > 0:
-            table.append_to_storage(buffer_df, TableType.TEMP)
+            temp_table.append_to_storage(buffer_df, self.ppss)
```
Comment (idiom-bytes) [inline pdr_backend/lake/gql_data_factory.py:228]: so much ppss being passed around...

**Review by idiom-bytes — APPROVED** [review]
I think this is overall a great improvement, however I'm seeing ppss being passed around so much and I wonder if there is a better pattern.... like a decorator or something else, such that the ppss is passed around by default, and can just be accessed wherever needed...

I have created a ticket to address this later.
https://github.com/oceanprotocol/pdr-backend/issues/1108

**Comment by calina-c** [comment]
> I think this is overall a great improvement, however I'm seeing ppss being passed around so much and I wonder if there is a better pattern.... like a decorator or something else, such that the ppss is passed around by default, and can just be accessed wherever needed...
> 
> I have created a ticket to address this later. #1108

I agree. I will address this later.

**Pull request merged**